### PR TITLE
Suggest fix for incomaptiabilty with windows file system

### PR DESCRIPTION
**Pull request by mki98**
Added a comment with a solution for windows users who might face a problem while creating a new file.

## Is this issue already raised? Yes/No
No

### CODE REVIEW

**Comment by ishtms** [comment]
Thanks @mki98 for raising this up. That was needed. I'll add some comments on what needs to be changed with this PR, if you can update the code, that would be great.

File: chapters/ch04.2-writing-logs.md [hunk 1]
```diff
@@ -411,6 +411,13 @@ class Logger {
 
     async init() {
         const file_name = this.#config.file_prefix + new Date().toISOString().replace(/\..+/, "") + ".log";
+
+        /**Expcted filename looks like this -> "LogTar_2023-08-26T14:46:11.log" this works well with Linux
+        * but it thorws an error if it is Windows because windows doesn't allow the colon ":" in filename
+        * So we could write this line an replace the dots and clolons with a -
+        * const file_name = const file_name=this.#config.file_prefix+ new Date().toISOString().replace(/[\..+::+]/,"-")+".log";
+        * Expcted filename looks like this -> "LogTar_2023-08-26T14-46-11.log" 
+        */
```
Comment (ishtms) [inline chapters/ch04.2-writing-logs.md:420]: We can remove the comment probably, and simplify the regex to be `[\.:]+`
So, the line
```js
const file_name = this.#config.file_prefix + new Date().toISOString().replace(/\..+/, "") + ".log";
```
becomes
```js
const file_name = this.#config.file_prefix + new Date().toISOString().replace(/[\.:]+/, "-") + ".log";
```
I'll do the rest, i.e add an extra line explaining about the `:` in the  - [A small primer to regular expressions](https://github.com/ishtms/learn-nodejs-hard-way/blob/master/chapters/ch04.2-writing-logs.md#a-small-primer-on-regular-expressions) section as well as explaining the `[]` regex operator.

**Review by ishtms — CHANGES_REQUESTED** [review]
If these changes are merged, I can do the rest of updating the point 4.

File: chapters/ch04.2-writing-logs.md (hunk 1 above)
Comment (mki98) [reply chapters/ch04.2-writing-logs.md:420]: Thank you for your feedback and guidance, @ishtms . I'm fully prepared to address the changes you've highlighted in the comments

**Review by mki98 — COMMENTED** [review]
Applied the requested changes and renamed all log files to be compatible with windows

**Review by ishtms — APPROVED** [review]
Thank you @mki98. Appreciate changing the `code/` directory too! Cheers 👍🏻

**Pull request merged**